gomint-server/src/main/java/io/gomint/server/network/packet/PacketStartGame.java

```diff
@@ -75,66 +79,86 @@
-        buffer.writeBoolean(false);
-        buffer.writeBoolean(false);
-        buffer.writeBoolean(false);
-        buffer.writeBoolean(false);
-
-        buffer.writeString(this.levelId);
-        buffer.writeString(this.worldName);
-        buffer.writeString(this.templateName);
-        buffer.writeBoolean(this.unknown1);
-        buffer.writeLLong(this.currentTick);
-        buffer.writeSignedVarInt(this.enchantmentSeed);
+        buffer.writeSignedVarInt( this.seed );
+        buffer.writeSignedVarInt( this.dimension );
+        buffer.writeSignedVarInt( this.generator );
+        buffer.writeSignedVarInt( this.worldGamemode );
+        buffer.writeSignedVarInt( this.difficulty );
+        buffer.writeSignedVarInt( (int) this.spawn.getX() );
+        buffer.writeUnsignedVarInt( (int) this.spawn.getY() );
+        buffer.writeSignedVarInt( (int) this.spawn.getZ() );
+        buffer.writeBoolean( this.hasAchievementsDisabled );
+        buffer.writeSignedVarInt( this.dayCycleStopTime );
+        buffer.writeSignedVarInt( this.eduEditionOffer );
+        buffer.writeBoolean( true ); // This is hasEduModeEnabled, we default to false until we have all EDU stuff in
+        buffer.writeLFloat( this.rainLevel );
+        buffer.writeLFloat( this.lightningLevel );
+        buffer.writeBoolean( false );
+        buffer.writeBoolean( this.isMultiplayerGame );
+        buffer.writeBoolean( this.hasLANBroadcast );
+        buffer.writeSignedVarInt( 3 );
+        buffer.writeSignedVarInt( 3 );
+        buffer.writeBoolean( this.commandsEnabled );
+        buffer.writeBoolean( this.isTexturePacksRequired );
+        writeGamerules( this.gamerules, buffer );
+        buffer.writeBoolean( this.hasBonusChestEnabled );
+        buffer.writeBoolean( this.hasStartWithMapEnabled );
+        buffer.writeSignedVarInt( this.defaultPlayerPermission );
+        buffer.writeInt( 32 );
+        buffer.writeBoolean( false );
+        buffer.writeBoolean( false );
+        buffer.writeBoolean( false );
+        buffer.writeBoolean( false );
+        buffer.writeBoolean( false );
+        buffer.writeBoolean( false );
+        buffer.writeBoolean( false );
+
+        buffer.writeString( this.vanillaVersion );
+        buffer.writeString( this.levelId );
+        buffer.writeString( this.worldName );
+        buffer.writeString( this.templateId );
+        buffer.writeBoolean( this.isTrial );
+        buffer.writeBoolean( this.movementServerAuthoritative );
+        buffer.writeLLong( this.currentTick );
+        buffer.writeSignedVarInt( this.enchantmentSeed );
 
         // Write palette data
-        byte[] data = BlockRuntimeIDs.getPacketCache();
-        buffer.writeBytes(data);
+        // byte[] data = BlockRuntimeIDs.getPacketCache();
+        // buffer.writeBytes( data );
+
+        // This is a temporary solution just to let people join and test
+        try {
+            InputStream inputStream = AssetsLibrary.class.getResourceAsStream( "/blocks.nbt" );
```

Comment:
From which version of the game is this dumped from?

Reply:
It's the PocketMine states Base64 encoded, latest version

Reply:
Basically is pocketmine states base64 encoded, latest version